src/tests/end-to-end/tests/details-view/headings.test.ts

```diff
@@ -0,0 +1,87 @@
+import { enableHighContrast } from '../../common/enable-high-contrast';
+import { Page } from '../../common/page';
+import { scanForAccessibilityIssues } from '../../common/scan-for-accessibility-issues';
+
+describe('Headings Page', () => {
+    describe('Normal mode', () => {
+        let browser: Browser;
+        let targetTabId: number;
+        let headingsPage: Page;
+
+        beforeAll(async () => {
+            browser = await launchBrowser({ suppressFirstTimeDialog: true });
+            targetTabId = (await browser.setupNewTargetPage()).tabId;
+            headingsPage = await openHeadingsPage(browser, targetTabId);
+        });
+
+        afterAll(async () => {
+            if (browser) {
+                await browser.close();
+                browser = undefined;
+            }
+        });
+
+        it('should pass accessibility validation', async () => {
+            const results = await scanForAccessibilityIssues(headingsPage, GuidanceContentSelectors.detailsContent);
+            expect(results).toHaveLength(0);
+        });
+    });
+
+    describe('High contrast mode', () => {
+        let browser: Browser;
+        let targetTabId: number;
+        let headingsPage: Page;
+
+        beforeAll(async () => {
+            browser = await launchBrowser({ suppressFirstTimeDialog: true });
+
+            targetTabId = (await browser.setupNewTargetPage()).tabId;
+            await setupHighContrastMode();
+
+            headingsPage = await openHeadingsPage(browser, targetTabId);
+        });
+
+        afterAll(async () => {
+            if (browser) {
+                await browser.close();
+                browser = undefined;
+            }
+        });
+
+        it('should pass accessibility validation', async () => {
+            const results = await scanForAccessibilityIssues(headingsPage, GuidanceContentSelectors.detailsContent);
+            expect(results).toHaveLength(0);
+        });
+
+        async function setupHighContrastMode(): Promise<void> {
+            const tempDetailsViewPage = await browser.newExtensionDetailsViewPage(targetTabId);
+            await enableHighContrast(tempDetailsViewPage);
+            await tempDetailsViewPage.close();
+        }
+    });
+
+    async function openHeadingsPage(browser: Browser, targetTabId: number): Promise<Page> {
+        const popupPage = await browser.newExtensionPopupPage(targetTabId);
+
+        let detailsViewPage: Page;
+
+        await Promise.all([
+            browser.waitForPageMatchingUrl(await browser.getDetailsViewPageUrl(targetTabId)).then(page => (detailsViewPage = page)),
+            popupPage.clickSelector(popupPageElementIdentifiers.launchPadAssessmentButton),
+        ]);
+
+        await detailsViewPage.waitForSelector(detailsViewSelectors.testNavArea);
+
+        await detailsViewPage.clickSelector(GuidanceContentSelectors.headingsNav);
+        await detailsViewPage.waitForSelectorToDisappear(DetailsViewCommonSelectors.loadingSpinner);
```

Comment:
I'd recommend waiting for a known element to appear, rather than the loading spinner to disappear; it isn't necessarily true that puppeteer will see the "spinner disappear + content appear" as an atomic action

Reply:
I was kind of questioning my implementation of that as well, so it was good to get a second opinion. I found that I can instead wait for selector `.assessment-instance-textContent` to appear, which would be one of the items in the Instances list.